### DISCUSSION TITLE
List all commands available within Supported Software Documentation

### DISCUSSION THREAD

**Discussion by ciscorucinski** [actions/virtual-environments#8135] [Ideas]
I have been checking out the [Supported Software](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-software) more often. This is where I knew that I could use the `jq` command without having to install it within my workflow file.

However, while looking at how to do things online, I noticed that I needed access to `seq` or `tr`, and it was my assumption that those were not supported software as they are not listed within the documentation at all. So I was trying to find alternative ways of doing these commands without installing them. _**However, I recently realized that those commands come from `coreutils` which is supported software.**_

I would rather prefer to google search `github actions supported software`, find the [link](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md) which shows me what is available, and search for `seq` or `ping` and see that those are in fact available to me. 

However, knowing that each installed package could have multiple commands available to them, if I wanted to easily know if it was supported, I would have to do a more lengthy deep dive to find a command's  package and then see if that package is available with my runner (add on top of that version differences)

**Comment by vpolikarpov-akvelon** [comment]
Hey @ciscorucinski. You can collect this information yourself by listing all executable files in each directory referenced in `$PATH`. For example, you may use this snippet on Ubuntu-based runner:

```yaml
- run: |
    IFS=:
    for dir in $PATH
    do
        for file in $dir/*
        do
            if [ -x $file ]
            then
                echo $file
            fi
        done
    done
```

I don't think there is a point to put a list of all available commands in documentation. Taking into account the overall amount of software that is installed on runners this list would be insanely huge.
